include/bzfsAPI.h

```diff
@@ -2000,11 +2000,20 @@ class bz_CustomZoneObject
     bool box;
     float xMax,xMin,yMax,yMin,zMax,zMin,radius,rotation;
 
+    // Half height and width only for boxes
+    float hh;
+    float hw;
+
+    // Center of the zone
+    float cX;
+    float cY;
+
+    // Used for rotation
+    float cos_val;
+    float sin_val;
+
```

Comment:
Do we want to make these to be public? The previous `xMax` and friends were public kinda by mistake (blame younger me). Maybe `protected` for these instead?

Reply:
Those attributes are used in a function that is not a member (the one to generate random point in a zone). I wonder why this function is not a member.

Reply:
Ah you're right, I missed that fact! It's partly the legacy design for the API *and* it has to do with compiled libs. I have a pretty poor understanding of things, but I believe that changing things inside of a class (e.g. adding new functions or changing its signature) could cause plugins compiled with older versions of bzfs to crash at runtime because offsets in memory addresses could be different. But if it's compiled this way, the offsets of memory addresses should be relatively safe to access